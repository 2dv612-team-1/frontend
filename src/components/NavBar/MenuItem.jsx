import styled from "styled-components";
import { Link } from "react-router-dom";
import colors from "../../constants/colors";

const MenuItem = styled(Link)`
  margin: 0 10px 0 10px;
  font-size: 18px;
  color: ${colors.secondary};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

export default MenuItem;
