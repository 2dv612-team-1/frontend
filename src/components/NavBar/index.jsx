import React from "react";
import PropTypes from "prop-types";
import MenuItem from "./MenuItem";
import Container from "./Container";

const defaultProps = {
  role: ""
};

const propTypes = {
  role: PropTypes.string
};

const NavBar = ({ role }) => (
  <Container>
    <MenuItem to="/">home</MenuItem>
    {role === "" ? (
      <div>
        <MenuItem to="/register/customer">Register</MenuItem>
        <MenuItem to="/login">Login</MenuItem>
        <MenuItem to="/adm">Admin</MenuItem>
      </div>
    ) : null}
    {role === "admin" ? (
      <div>
        <MenuItem to="/register/company">Register</MenuItem>
        <MenuItem to="/companies">Companies</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
    {role === "company" ? (
      <div>
        <MenuItem to="/register/representative">Register</MenuItem>
        <MenuItem to="/representatives">Representatives</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
    {role === "representative" ? (
      <div>
        <MenuItem to="/products">Products</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
    {role === "consumer" ? (
      <div>
        <MenuItem to="/products">Products</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
  </Container>
);

NavBar.defaultProps = defaultProps;
NavBar.propTypes = propTypes;

export default NavBar;
