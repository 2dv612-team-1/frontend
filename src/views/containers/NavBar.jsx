import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import MenuItem from "../components/MenuItem";
import FlexBox from "../components/FlexBox";

const defaultProps = {
  role: ""
};

const propTypes = {
  role: PropTypes.string
};

const NavBar = ({ role }) => (
  <FlexBox>
    <MenuItem to="/">home</MenuItem>
    {role === "" ? (
      <div>
        <MenuItem to="/register/customer">Register</MenuItem>
        <MenuItem to="/login">Login</MenuItem>
        <MenuItem to="/adm">Admin</MenuItem>
      </div>
    ) : null}
    {role === "admin" ? (
      <div>
        <MenuItem to="/categories">Categories</MenuItem>
        <MenuItem to="/companies">Companies</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
    {role === "company" ? (
      <div>
        <MenuItem to="/representatives">Representatives</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
    {role === "representative" ? (
      <div>
        <MenuItem to="/products">Products</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
        <MenuItem to="/forum">Forum</MenuItem>
      </div>
    ) : null}
    {role === "consumer" ? (
      <div>
        <MenuItem to="/products">Products</MenuItem>
        <MenuItem to="/forum">Forum</MenuItem>
        <MenuItem to="/thread/new">New</MenuItem>
        <MenuItem to="/logout">Logout</MenuItem>
      </div>
    ) : null}
  </FlexBox>
);

const mapStateToProps = state => ({
  role: state.session.loggedInAs.role
});

NavBar.defaultProps = defaultProps;
NavBar.propTypes = propTypes;
export default connect(mapStateToProps)(NavBar);
