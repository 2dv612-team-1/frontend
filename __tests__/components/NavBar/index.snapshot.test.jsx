import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NavBar from "../../../src/components/NavBar";

describe("NavBar", () => {
  describe("When not loggedin", () => {
    it("should render correctly", () => {
      const tree = shallow(<NavBar role="" />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
  describe("When loggedin as admin", () => {
    it("should render correctly", () => {
      const tree = shallow(<NavBar role="admin" />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
  describe("When loggedin as company", () => {
    it("should render correctly", () => {
      const tree = shallow(<NavBar role="company" />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
  describe("When loggedin as representative", () => {
    it("should render correctly", () => {
      const tree = shallow(<NavBar role="representative" />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
});
