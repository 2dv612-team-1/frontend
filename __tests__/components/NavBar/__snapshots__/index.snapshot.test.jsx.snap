// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar When loggedin as admin should render correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    home
  </Styled(Link)>
  <div>
    <Styled(Link)
      to="/register/company"
    >
      register
    </Styled(Link)>
    <Styled(Link)
      to="/companies"
    >
      companies
    </Styled(Link)>
    <Styled(Link)
      to="/logout"
    >
      logout
    </Styled(Link)>
  </div>
</styled.div>
`;

exports[`NavBar When loggedin as company should render correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    home
  </Styled(Link)>
  <div>
    <Styled(Link)
      to="/login"
    >
      login
    </Styled(Link)>
    <Styled(Link)
      to="/register/customer"
    >
      register
    </Styled(Link)>
  </div>
  <div>
    <Styled(Link)
      to="/register/representative"
    >
      register
    </Styled(Link)>
    <Styled(Link)
      to="/representatives"
    >
      representatives
    </Styled(Link)>
  </div>
</styled.div>
`;

exports[`NavBar When loggedin as representative should render correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    home
  </Styled(Link)>
  <div>
    <Styled(Link)
      to="/login"
    >
      login
    </Styled(Link)>
    <Styled(Link)
      to="/register/customer"
    >
      register
    </Styled(Link)>
  </div>
</styled.div>
`;

exports[`NavBar When not loggedin should render correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    home
  </Styled(Link)>
  <div>
    <Styled(Link)
      to="/login"
    >
      login
    </Styled(Link)>
    <Styled(Link)
      to="/register/customer"
    >
      register
    </Styled(Link)>
  </div>
</styled.div>
`;
